js/modules/k6/http/http.go

```diff
@@ -21,101 +21,152 @@
+		args = append([]goja.Value{goja.Undefined()}, args...) // sigh
+		return mi.defaultClient.Request(http.MethodGet, url, args...)
+	})
+	mustExport("head", mi.defaultClient.getMethodClosure(http.MethodHead))
+	mustExport("post", mi.defaultClient.getMethodClosure(http.MethodPost))
+	mustExport("put", mi.defaultClient.getMethodClosure(http.MethodPut))
+	mustExport("patch", mi.defaultClient.getMethodClosure(http.MethodPatch))
+	mustExport("del", mi.defaultClient.getMethodClosure(http.MethodDelete))
+	mustExport("options", mi.defaultClient.getMethodClosure(http.MethodOptions))
+	mustExport("request", mi.defaultClient.Request)
+	mustExport("batch", mi.defaultClient.Batch)
+	mustExport("setResponseCallback", mi.defaultClient.SetResponseCallback)
+
+	mustExport("expectedStatuses", mi.expectedStatuses) // TODO: refactor?
+
+	// TODO: actually expose the default client as k6/http.defaultClient when we
+	// have a better HTTP API (e.g. proper Client constructor, an actual Request
+	// object, custom Transport implementations you can pass the Client, etc.)
+
+	return mi
 }
 
-//nolint: golint
-type HTTP struct {
-	TLS_1_0                            string `js:"TLS_1_0"`
-	TLS_1_1                            string `js:"TLS_1_1"`
-	TLS_1_2                            string `js:"TLS_1_2"`
-	TLS_1_3                            string `js:"TLS_1_3"`
-	OCSP_STATUS_GOOD                   string `js:"OCSP_STATUS_GOOD"`
-	OCSP_STATUS_REVOKED                string `js:"OCSP_STATUS_REVOKED"`
-	OCSP_STATUS_SERVER_FAILED          string `js:"OCSP_STATUS_SERVER_FAILED"`
-	OCSP_STATUS_UNKNOWN                string `js:"OCSP_STATUS_UNKNOWN"`
-	OCSP_REASON_UNSPECIFIED            string `js:"OCSP_REASON_UNSPECIFIED"`
-	OCSP_REASON_KEY_COMPROMISE         string `js:"OCSP_REASON_KEY_COMPROMISE"`
-	OCSP_REASON_CA_COMPROMISE          string `js:"OCSP_REASON_CA_COMPROMISE"`
-	OCSP_REASON_AFFILIATION_CHANGED    string `js:"OCSP_REASON_AFFILIATION_CHANGED"`
-	OCSP_REASON_SUPERSEDED             string `js:"OCSP_REASON_SUPERSEDED"`
-	OCSP_REASON_CESSATION_OF_OPERATION string `js:"OCSP_REASON_CESSATION_OF_OPERATION"`
-	OCSP_REASON_CERTIFICATE_HOLD       string `js:"OCSP_REASON_CERTIFICATE_HOLD"`
-	OCSP_REASON_REMOVE_FROM_CRL        string `js:"OCSP_REASON_REMOVE_FROM_CRL"`
-	OCSP_REASON_PRIVILEGE_WITHDRAWN    string `js:"OCSP_REASON_PRIVILEGE_WITHDRAWN"`
-	OCSP_REASON_AA_COMPROMISE          string `js:"OCSP_REASON_AA_COMPROMISE"`
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Default: mi.exports,
+		// TODO: add new HTTP APIs like Client, Request, etc. as named exports?
+	}
+}
 
-	responseCallback func(int) bool
+// TODO: deprecate these?
+func (mi *ModuleInstance) defineConstants() {
+	rt := mi.vu.Runtime()
+	mustAddProp := func(name, val string) {
+		err := mi.exports.DefineDataProperty(
+			name, rt.ToValue(val), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE,
+		)
+		if err != nil {
+			common.Throw(rt, err)
+		}
```

Comment:
I feel like this is the one use case for using `goja.Object` as the exports - being able to make things not act just as a simple map, as in this case not being able to overwrite this constants.

The question is should you not be able to overwrite this but be able to overwrite the functions ?
Should we return something different that is more robust in handling such cases 

See https://github.com/grafana/k6/pull/2241#issuecomment-977836690 (And above comments) for discussion on why I prefer `map[string]interface{}` for most cases

Reply:
Hmm :thinking: I can understand the `map[string]interface{}` preference, but goja will just convert that to an `Object` anyway. And I like being able to have constants be constant :sweat_smile: 

Do you know hot this should work with proper ES modules? If you have `export const something`, does that actually make it constant in the code that imports it?

Reply:
> If you have export const something, does that actually make it constant in the code that imports it? 

yes